src/realm/sync/subscriptions.cpp

```diff
@@ -0,0 +1,457 @@
+        auto sub_sets = tr->get_table(m_sub_set_keys->table);
+        m_sub_set_keys->state = lookup_and_validate_column(sub_sets, c_flx_sub_sets_state_field, type_Int);
+        m_sub_set_keys->error_str = lookup_and_validate_column(sub_sets, c_flx_sub_sets_error_str_field, type_String);
+        m_sub_set_keys->subscriptions =
+            lookup_and_validate_column(sub_sets, c_flx_sub_sets_subscriptions_field, type_LinkList);
+        if (!m_sub_set_keys->subscriptions) {
+            throw std::runtime_error("Flexible Sync metadata missing subscriptions table");
+        }
+
+        auto subs = sub_sets->get_opposite_table(m_sub_set_keys->subscriptions);
+        if (!subs->is_embedded()) {
+            throw std::runtime_error("Flexible Sync subscriptions table should be an embedded object");
+        }
+        m_sub_keys->table = subs->get_key();
+        m_sub_keys->created_at = lookup_and_validate_column(subs, c_flx_sub_created_at_field, type_Timestamp);
+        m_sub_keys->updated_at = lookup_and_validate_column(subs, c_flx_sub_updated_at_field, type_Timestamp);
+        m_sub_keys->query_str = lookup_and_validate_column(subs, c_flx_sub_query_str_field, type_String);
+        m_sub_keys->object_class_name = lookup_and_validate_column(subs, c_flx_sub_object_class_field, type_String);
+        m_sub_keys->name = lookup_and_validate_column(subs, c_flx_sub_name_field, type_String);
+    }
+}
+
+const SubscriptionSet SubscriptionStore::get_latest() const
+{
+    auto tr = m_db->start_read();
+    auto sub_sets = tr->get_table(m_sub_set_keys->table);
+    if (sub_sets->is_empty()) {
+        return SubscriptionSet(this, std::move(tr), Obj{});
+    }
+    auto latest_id = sub_sets->maximum_int(sub_sets->get_primary_key_column());
+    auto latest_obj = sub_sets->get_object_with_primary_key(Mixed{latest_id});
+    return SubscriptionSet(this, std::move(tr), std::move(latest_obj));
+}
+
+const SubscriptionSet SubscriptionStore::get_active() const
+{
+    auto tr = m_db->start_read();
+    auto sub_sets = tr->get_table(m_sub_set_keys->table);
+    if (sub_sets->is_empty()) {
+        return SubscriptionSet(this, std::move(tr), Obj{});
+    }
+
+    DescriptorOrdering descriptor_ordering;
+    descriptor_ordering.append_sort(SortDescriptor{{{sub_sets->get_primary_key_column()}}, {false}});
+    descriptor_ordering.append_limit(LimitDescriptor{1});
+    auto res = sub_sets->where()
+                   .equal(m_sub_set_keys->state, static_cast<int64_t>(SubscriptionSet::State::Complete))
+                   .find_all(descriptor_ordering);
+
+    if (res.is_empty()) {
+        tr->close();
+        return get_latest();
+    }
+    return SubscriptionSet(this, std::move(tr), res.get(0));
+}
+
+SubscriptionSet SubscriptionStore::get_mutable_by_version(int64_t version_id)
+{
+    auto tr = m_db->start_write();
+    auto sub_sets = tr->get_table(m_sub_set_keys->table);
+    auto obj_key = sub_sets->get_objkey_from_primary_key(Mixed{version_id});
```

Comment:
that method is only for the sync client, the one you're looking for is `find_primary_key`